pkgs/test/nixpkgs-check-by-name/src/nixpkgs_problem.rs

```diff
@@ -158,12 +195,101 @@ impl fmt::Display for NixpkgsProblem {
                     "pkgs.{package_name}: This attribute is not defined but it should be defined automatically as {}",
                     relative_package_file.display()
                 ),
-            NixpkgsProblem::WrongCallPackage { relative_package_file, package_name } =>
-                write!(
+            NixpkgsProblem::EmptyArgument { package_name, file, line, column, definition } =>
+                writedoc!(
                     f,
-                    "pkgs.{package_name}: This attribute is manually defined (most likely in pkgs/top-level/all-packages.nix), which is only allowed if the definition is of the form `pkgs.callPackage {} {{ ... }}` with a non-empty second argument.",
-                    relative_package_file.display()
+                    "
+                    - Because {} exists, the attribute `pkgs.{package_name}` must be defined like
+
+                        {package_name} = callPackage ./{} {{ /* ... */ }};
+
+                      Notably the second argument must not be empty, which is not the case.
+                      It is defined in {}:{} as
```

Comment:
I'll apply the rewording.

Regarding what to do next, I'd like to push that to another PR. I think it would be best to have documentation for resolving each of these errors and then linking to it from the errors.

Reply:
Though for this specific error it's a fairly straightforward improvement, which I now implemented with https://github.com/NixOS/nixpkgs/pull/290743/commits/7858f062884e2318dda212ab7cc1ceda871195b2 :)

Reply:
Nice! I like it a lot.